tools/run_block.py

```diff
@@ -98,8 +100,8 @@ def npc_to_dict(npc: NPC):
 
 def run_generator(block_generator: BlockGenerator, constants: ConsensusConstants, max_cost: int) -> List[CAT]:
 
-    args = create_generator_args(block_generator.generator_refs).first()
-    _, block_result = block_generator.program.run_with_cost(max_cost, 0, args)
```

Comment:
as far as I can tell, passing `0` as the first argument here means that the block generator does not have access to the clvm parser. if it tries to parse one of the block-ref arguments, presumably this would fail.

Reply:
Yes, I believe that's true – the standard says a (buggy and therefore rather useless) piece of code that deserializes clvm is passed to the generator. We can't change it without a hard fork.

So I will write a simple drop-in block generator that fails if this argument is wrong. We can use this for validation here and possibly other places... essentially "copy-protection" to ensure that the ROM is compatible.

Reply:
I see. that explains why nothing seems to be failing, despite passing 0 here

Reply:
https://gist.github.com/richardkiss/e4a0e6ceb97489f7a9f1d3938b1a0d43

Reply:
there should be a test against that block, yes. More importantly, the main blockchain should be tested against it.